include/swift/AST/DiagnosticsSIL.def

```diff
@@ -371,6 +371,23 @@ ERROR(possible_return_before_yield, none,
 
 NOTE(one_yield, none, "yield along one path is here", ())
 
+NOTE(false_branch_doesnt_yield, none,
+    "must yield when the condition is false", ())
+
+NOTE(true_branch_doesnt_yield, none,
+    "must yield when the condition is true", ())
+
+NOTE(switch_case_doesnt_yield, none, "must yield in the case '%0'", (StringRef))
+
+NOTE(default_case_doesnt_yield, none, "must yield in the default case", ())
+
+NOTE(switch_value_case_doesnt_yield, none, "must yield in case %0",
```

Comment:
You can have this say "in the nth case" using llvm::getOrdinalSuffix(). (See my later comment)